### ISSUE TITLE
Linear Genome Setting

### ISSUE THREAD

**Issue by paolij** [PoonLab/OpenRDP#71]
Hello Art,

I successfully installed openRDP. How can I change the settings to analyze a linear genome (for viruses)? Thank you!

**Comment by ArtPoon** [comment]
Sorry it took me so long to get to this issue, we've been on a holiday break here.  I'm not sure what you mean by linear genome - this should not be substantially different from analyzing an alignment of partial genome (i.e., gene) sequences.  If you are asking about how to modify parameter settings for longer input sequences, I'd recommend referring to the original RDP documentation.

**Comment by darrenpmartin** [comment]
I think he is referring to the original RDP setting where sequences are
analysed as being circular or linear.  For circular sequences the window
wraps at the ends of the sequences.

**Comment by paolij** [comment]
Hi Art,

No worries, hope you had a joyful holiday season!

Yes, I was wondering in OpenRDP how to specify linear genomes as opposed to default circular to analyze viral genomes.

I was able to install OpenRDP in UF's supercomputer and write a script which ran sucessfully, however I realized the output options I require for my project are not yet available in OpenRDP (for example having the program remove sections of sequences with recombination evidence from my fasta file). I will stick with the desktop RDP5 version for now.

Thank you for your help!

Best wishes,

Julia Paoli, MSc (she/her)

Mavian Lab

Emerging Pathogens Institute
Department of Pathology
College of Medicine
University of Florida
________________________________

**Comment by ArtPoon** [comment]
Yes, OpenRDP does not have the complete functionality of RDP5, such as removal of recombinant sections.  My plan is to have one of my lab members work on this problem, but it is not covered by any of my research funding, so I cannot assign much priority to this task.

**Comment by paolij** [comment]
No worries! OpenRDP is a great initiative. Looking forward to its future capabilities!

Best wishes,

Julia Paoli, MSc (she/her)

Mavian Lab

Emerging Pathogens Institute
Department of Pathology
College of Medicine
University of Florida
________________________________

**Comment by ArtPoon** [comment]
We don't currently provide any settings for circular versus linear genomes in the driver script.  I'll have to dig into the main routines to check what the defaults really are.  This might be an easy fix, or it might not.

**Comment by ArtPoon** [comment]
@WilliamZekaiWang can you please check which setting we are using by default?

**Comment by ArtPoon** [comment]
@wqyang421 can you please update this issue with your findings?  Thanks!

**Comment by wqyang421** [comment]
Sorry for the delay. I've reviewed the code, and it doesn't differentiate between circular and linear genomes in its current settings. I also searched the RDP methods, which focus on recombination rather than physical structure. According to the RDP4 Manual, if recombination breakpoint distributions are of interest, it would almost always be best to inform the program whether the sequences being analyzed are linear or circular. Thus, we might still need a parameter to handle that in the future.
